### PR TITLE
DOC/CLN: Timezone section in timeseries.rst

### PR DESCRIPTION
**Pull request by mroeschke**
- Shortened some of the examples to show the minimum, necessary functionality
- Removed some redundant examples
- Refactored the ambiguous times handling section
- Showcase `tz_localize` more in the beginning of the section & noted that pandas doesn't support `datetime.timezone` objects yet

### CODE REVIEW

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -2129,11 +2129,13 @@ These can easily be converted to a ``PeriodIndex``:
 Time Zone Handling
 ------------------
```
Comment (jreback) [inline doc/source/timeseries.rst]: I would split timezones off into a separate section entirely (but can do that in a separate PR)

File: doc/source/timeseries.rst [hunk 2]
```diff
@@ -2145,94 +2147,86 @@ By default, pandas objects are time zone unaware:
    rng = pd.date_range('3/6/2012 00:00', periods=15, freq='D')
    rng.tz is None
 
-To supply the time zone, you can use the ``tz`` keyword to ``date_range`` and
-other functions. Dateutil time zone strings are distinguished from ``pytz``
-time zones by starting with ``dateutil/``.
+To localize these dates to a time zone, you can use the ``tz_localize`` method
```
Comment (jreback) [inline doc/source/timeseries.rst]: maybe have some definitions here, you jump right into localization w/o explaining what it is

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/timeseries.rst [hunk 3]
```diff
@@ -2145,94 +2147,86 @@ By default, pandas objects are time zone unaware:
...
-   rng_utc = pd.date_range('3/6/2012 00:00', periods=10, freq='D',
+   rng_utc = pd.date_range('3/6/2012 00:00', periods=3, freq='D',
                            tz=dateutil.tz.tzutc())
    rng_utc.tz
 
-Note that the ``UTC`` timezone is a special case in ``dateutil`` and should be constructed explicitly
-as an instance of ``dateutil.tz.tzutc``. You can also construct other timezones explicitly first,
-which gives you more control over which time zone is used:
+Note that the ``UTC`` time zone is a special case in ``dateutil`` and should be constructed explicitly
+as an instance of ``dateutil.tz.tzutc``. You can also construct other the time
```
Comment (TomAugspurger) [inline doc/source/timeseries.rst]: Typo: maybe remove "the" before "the time zones"

File: doc/source/timeseries.rst [hunk 4]
```diff
@@ -2145,94 +2147,86 @@ By default, pandas objects are time zone unaware:
...
-
-   ts = pd.Series(np.random.randn(len(rng)), rng)
-
-   ts_utc = ts.tz_localize('UTC')
-   ts_utc
-
-Again, you can explicitly construct the timezone object first.
-You can use the ``tz_convert`` method to convert pandas objects to convert
-tz-aware data to another time zone:
+To convert a pandas objects from one time zone to another,
```
Comment (TomAugspurger) [inline doc/source/timeseries.rst]: objects -> object.

Maybe "To convert a *timezone-aware* pandas object".

File: doc/source/timeseries.rst (hunk 1 above)
Comment (mroeschke) [reply doc/source/timeseries.rst]: Agreed. I think there's enough content in `timeseries.rst` to split into `Timestamp` (and timezones), `Timedelta`, `Period` and `Offset`. Will work on those in separate PRs

**Comment by mroeschke** [comment]
Thanks. Addressed the comments.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @TomAugspurger could merge to 0.24.1 if you'd like.

**Comment by TomAugspurger** [comment]
I think we agreed that backporting doc fixes wasn't worth the hassle.
